### PR TITLE
apache-arrow 10.0.0

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

File: Formula/apache-arrow.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 class ApacheArrow < Formula
   desc "Columnar in-memory analytics layer designed to accelerate big data"
   homepage "https://arrow.apache.org/"
-  url "https://downloads.apache.org/arrow/arrow-9.0.0/apache-arrow-9.0.0.tar.gz"
+  #url "https://downloads.apache.org/arrow/arrow-10.0.0/apache-arrow-10.0.0.tar.gz"
```
Comment (nealrichardson) [inline Formula/apache-arrow.rb]: probably can't do that

```suggestion
  url "https://downloads.apache.org/arrow/arrow-10.0.0/apache-arrow-10.0.0.tar.gz"
```

File: Formula/apache-arrow.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ class ApacheArrow < Formula
   depends_on "zstd"
 
   def install
-    ENV.cxx11
+    ENV.cxx17
```
Comment (nealrichardson) [inline Formula/apache-arrow.rb]:
```suggestion
```

**Comment by jeroen** [comment]
Maybe you need to pass `_LIBCPP_DISABLE_AVAILABILITY` like here? https://github.com/apache/arrow/pull/14105
How did you make it work on your internal MacOS 10.13 CI?

File: Formula/apache-arrow.rb [hunk 3]
```diff
@@ -22,7 +22,6 @@ class ApacheArrow < Formula
   depends_on "zstd"
 
   def install
-    ENV.cxx11
     args = %W[
       -DARROW_COMPUTE=ON
       -DARROW_CSV=ON
```
Comment (nealrichardson) [inline Formula/apache-arrow.rb:27]: Yep, I think this is needed. I don't see any other changes in https://github.com/apache/arrow/blame/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow.rb that aren't here. 

```suggestion
      -DARROW_CSV=ON
      -DARROW_CXXFLAGS="-D_LIBCPP_DISABLE_AVAILABILITY"
```

**Comment by nealrichardson** [comment]
Looks like it built that time but `audit` failed (expected since this is an RC) and so did `test` (IDK if that's real or not, we don't run `brew test` in our CI I don't think).

**Comment by jeroen** [comment]
Yeah the bottle is there, looks good. Not sure what's with that test, looks like some CI problem :/

**Comment by jeroen** [comment]
I'm slightly worried the error is real though. Because we are using this macro to override this error

```
error: call to unavailable function 'get': introduced in macOS 10.14
```

But because this is actually MacOS 10.13 that could mean the resulting binary cannot be not run on this system, which would explain a vague error in the tests.

I'll deploy this to staging and do some testing before submitting to CRAN...

**Comment by jeroen** [comment]
Added `-D_LIBCPP_DISABLE_AVAILABILITY` to the test and now it passes on CI 🤞

**Comment by jeroen** [comment]
All seems good to go on my High-Sierra VM:

```
R CMD check arrow_10.0.0.tar.gz --no-manual
* using log directory ‘/Users/builder/Desktop/arrow/arrow.Rcheck’
* using R version 4.2.0 (2022-04-22)
* using platform: x86_64-apple-darwin17.0 (64-bit)
* using session charset: UTF-8
* using option ‘--no-manual’
* checking for file ‘arrow/DESCRIPTION’ ... OK
* this is package ‘arrow’ version ‘10.0.0’
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘arrow’ can be installed ... OK
* checking installed package size ... NOTE
  installed size is 47.6Mb
  sub-directories of 1Mb or more:
    R      5.0Mb
    libs  41.4Mb
* checking package directory ... OK
* checking ‘build’ directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking line endings in shell scripts ... OK
* checking line endings in C/C++/Fortran sources/headers ... OK
* checking line endings in Makefiles ... OK
* checking compilation flags in Makevars ... OK
* checking for GNU extensions in Makefiles ... OK
* checking for portable use of $(BLAS_LIBS) and $(LAPACK_LIBS) ... OK
* checking use of PKG_*FLAGS in Makefiles ... OK
* checking compiled code ... OK
* checking installed files from ‘inst/doc’ ... OK
* checking files in ‘vignettes’ ... OK
* checking examples ... OK
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’
 OK
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in ‘inst/doc’ ... OK
* checking running R code from vignettes ...
  ‘arrow.Rmd’ using ‘UTF-8’... OK
  ‘dataset.Rmd’ using ‘UTF-8’... OK
  ‘developing.Rmd’ using ‘UTF-8’... OK
  ‘flight.Rmd’ using ‘UTF-8’... OK
  ‘fs.Rmd’ using ‘UTF-8’... OK
  ‘install.Rmd’ using ‘UTF-8’... OK
  ‘python.Rmd’ using ‘UTF-8’... OK
 NONE
* checking re-building of vignette outputs ... OK
* DONE

Status: 1 NOTE
```

**Pull request merged**